File: arena/src/main/java/org/zenika/arena/repository/BattleRepository.java
package org.zenika.arena.repository;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.zenika.arena.entity.BattleEntity;

import java.util.List;

public interface BattleRepository extends MongoRepository<BattleEntity, String> {

    List<BattleEntity> findAllByWinnerIsNull();

}

File: arena/src/main/kotlin/org/zenika/arena/repository/BattleRepository.kt
package org.zenika.arena.repository;

import org.springframework.data.mongodb.repository.MongoRepository
import org.zenika.arena.entity.BattleEntity;


interface BattleRepository: MongoRepository<BattleEntity, String> {

    fun findAllByWinnerIsNull(): List<BattleEntity>

}
